tests/test_reply_function_async.js

```diff
@@ -125,3 +93,56 @@ test('subsequent calls to the reply callback are ignored', async t => {
   t.is(statusCode, 201)
   t.equal(body, 'one')
 })
+
+test('reply can take a direct async function without a callback', async t => {
+  const scope = nock('http://example.com')
+    .get('/')
+    .reply(200, async (path, requestBody) => {
+      t.equal(path, '/')
+      t.equal(requestBody, '')
```

Comment:
It would make a better assertion if you passed something more specific in, probably both for the path and the body.